js/views/media-frame.js

```diff
@@ -44,6 +45,19 @@ var mediaFrame = postMediaFrame.extend( {
 
 	},
 
+	events: function() {
+		return _.extend( {}, postMediaFrame.prototype.events, {
+			'click'    : 'resetMediaController',
+		} );
+	},
+
+	resetMediaController: function( event ) {
+		if ( $( event.target ).hasClass( 'media-menu-item' ) ) {
+			this.mediaController.reset();
+			this.contentRender( 'shortcode-ui', 'insert' );
+		}
```

Comment:
Figured it out f925bbf9b628b2e710e1bf103bd29b7244b5034e 033d398d512a3bb187e77053a1c6a7d4fe99715a